driver/chain_syncer/chain_syncer_test.go

```diff
@@ -46,15 +72,71 @@ func (s *ChainSyncerTestSuite) TestSync() {
 	s.Nil(s.s.Sync(head))
 }
 
-// func (s *ChainSyncerTestSuite) TestSyncTriggerBeaconSync() {
-// 	s.s.p2pSyncVerifiedBlocks = true
-// 	s.s.state.setLatestVerifiedBlockHash(common.Hash{})
-// }
+func (s *ChainSyncerTestSuite) TestAheadOfProtocolVerifiedHead2() {
+	s.TakeSnapshot()
+	// propose a couple blocks
+	testutils.ProposeAndInsertEmptyBlocks(&s.ClientTestSuite, s.p, s.s.calldataSyncer)
+
+	// NOTE: need to prove the proposed blocks to be verified, writing helper function
+
+	// generate transactopts to interact with TaikoL1 contract with.
+	privKey, err := crypto.ToECDSA(common.Hex2Bytes(os.Getenv("L1_PROVER_PRIVATE_KEY")))
+	s.Nil(err)
+	opts, err := bind.NewKeyedTransactorWithChainID(privKey, s.RpcClient.L1ChainID)
+	s.Nil(err)
+
+	head, err := s.RpcClient.L1.HeaderByNumber(context.Background(), nil)
+	s.Nil(err)
+
+	l2Head, err := s.RpcClient.L2.HeaderByNumber(context.Background(), nil)
+	s.Nil(err)
+
+	log.Info("L1HeaderByNumber head: %v\n", head.Number)
+	// (equiv to s.state.GetL2Head().Number)
+	log.Info("L2HeaderByNumber head: %v\n", l2Head.Number)
+	log.Info("LatestVerifiedBlock number: %v\n", s.s.state.GetLatestVerifiedBlock().ID.Uint64())
+
+	// increase evm time to make blocks verifiable.
+	var result uint64
+	s.Nil(s.RpcClient.L1RawRPC.CallContext(context.Background(), &result, "evm_increaseTime", 2000))
```

Comment:
Lets not sure hard-coded `2000` but the real cooldown time config from `s.s.rpc.TaikoL1.GetConfig()`?